lib/Sparql/Client.php

```diff
@@ -1,395 +1,468 @@
+            return $result;
+        }
+    }
+
+    /**
+     * This is a low-level version of the query() function where the Accept header is
+     * provided explicitely, as well as the default and named graphs URIs.
+     *
+     * The result is the body of the raw HTTP response, not an EasyRdf_Sparql_Result or
+     * EasyRdf_Graph like in function request().
+     *
+     * @param string $query
+     *            The query string to be executed
+     * @param string $accept
+     *            Gives the value of the Accept HTTP header
+     * @param string $defaultGraphUri
+     *            Used as the default-graph-uri header
+     * @param string $namedGraphUri
+     *            Used as the named-graph-uri header
+     * @param string $usingGraphUri
+     *            Used as the using-graph-uri header
+     * @param string $usingNamedGraphUri
+     *            Used as the using-named-graph-uri header
+     * @return Http\Response the raw HTTP response
+     *        
+     * @ignore
+     */
+    public function queryRaw($query, $accept, $defaultGraphUri = null, $namedGraphUri = null, $usingGraphUri = null, $usingNamedGraphUri = null)
+    {
+        // Check for undefined prefixes
+        $prefixes = '';
+        foreach (RdfNamespace::namespaces() as $prefix => $uri) {
+            if (strpos($query, "{$prefix}:") !== false and strpos($query, "PREFIX {$prefix}:") === false) {
+                $prefixes .= "PREFIX {$prefix}: <{$uri}>\n";
+            }
+        }
+        
+        $client = Http::getDefaultHttpClient();
+        $client->resetParameters();
+
+        // Tell the server which response formats we can parse
+        $client->setHeaders('Accept', $accept);
+
+        $encodedQuery = 'query=' . urlencode($prefixes . $query);
+
+        // In the "query via URL-encoded POST", the graph URIs are passed in the request message body
+        if ($defaultGraphUri)
+            $encodedQuery = $encodedQuery . '&default-graph-uri=' . urlencode($defaultGraphUri);
+        if ($namedGraphUri)
+            $encodedQuery = $encodedQuery . '&named-graph-uri=' . urlencode($namedGraphUri);
+
+        if ($usingGraphUri)
+            $encodedQuery = $encodedQuery . '&using-graph-uri=' . urlencode($usingGraphUri);
+        if ($usingNamedGraphUri)
+            $encodedQuery = $encodedQuery . '&using-named-graph-uri=' . urlencode($usingNamedGraphUri);
+
+        $client->setUri($this->queryUri);
+        $client->setMethod('POST');
+        $client->setRawData($encodedQuery);
+        $client->setHeaders('Content-Type', 'application/x-www-form-urlencoded');
+
+        $response = $client->request();
+        if ($response->getStatus() == 204) {
+            // No content
+            return $response;
+        } elseif ($response->isSuccessful()) {
+            return $response;
+        } else
+            throw new Exception("HTTP request for SPARQL query failed: " . $response->getBody());
+    }
+}

```

Comment:
Why not do the change via browser? That should keep the file formatting.

**Suggestion:** Open the original file in the browser https://raw.githubusercontent.com/easyrdf/easyrdf/master/lib/Sparql/Client.php and copy its content into the same file in your fork (opened on Github also via browser). Afterwards add the new function in the end. Save it. That should keep the formatting. *I am afraid it may get rejected because of the formatting issues.*